src/blockMonitor/blockCache.ts

```diff
@@ -72,22 +71,16 @@ export class BlockCache<TBlock extends IBlockStub> implements ReadOnlyBlockCache
      */
     constructor(public readonly maxDepth: number) {}
 
-    // Removes all info related to a block in blockStubsByHash
-    private removeBlock(blockHash: string) {
-        const block = this.blockStubsByHash.get(blockHash);
-        if (!block) {
-            // This would signal a bug
-            throw new ApplicationError(`Block with hash ${blockHash} not found, but it was expected.`);
-        }
-
-        this.blockStubsByHash.delete(blockHash);
-    }
-
     // Remove all the blocks that are deeper than maxDepth, and all connected information.
     private prune() {
         while (this.pruneHeight < this.minHeight) {
             for (const hash of this.blockHashesByHeight.get(this.pruneHeight) || []) {
-                this.removeBlock(hash);
+                const deleted = this.blocksByHash.delete(hash) || this.pendingBlocksByHash.delete(hash);
+
+                if (!deleted) {
```

Comment:
I don't like doing `if (this.blocksByHash.delete(hash) || this.pendingBlocksByHash.delete(hash)) {...}` because I think it's easy to miss that there are side effect in the condition.

Reply:
ah no, what I meant by this was just doing
```
// This would signal a bug
if(!deleted) throw new ApplicationError(`Tried to delete block with hash ${hash}, but it does not exist.`); 
```

vs 

```
if (!deleted) {
 // This would signal a bug
 throw new ApplicationError(`Tried to delete block with hash ${hash}, but it does not exist.`);
}
```

Reply:
Changed similarly to the style you suggest.